tests/test_enki.py

```diff
@@ -740,6 +762,47 @@ def update_circulation(self, since):
         eq_([1,2], importer.processed)
 
     def test_update_circulation(self):
+        # update_circulation() makes two-hour slices out of time
+        # between the previous run and now, and passes each slice into
+        # _update_circulation, keeping track of the total number of
+        # circulation events encountered.
+        class Mock(EnkiImport):
+
+            def __init__(self, *args, **kwargs):
+                super(Mock, self).__init__(*args, **kwargs)
+                self._update_circulation_called_with = []
+                self.sizes = [1,2]
+
+            def _update_circulation(self, start, end):
+                # Pretend that one circulation event was discovered
+                # during the given time span.
+                self._update_circulation_called_with.append((start, end))
+                return self.sizes.pop()
+
+        # Call update_circulation() on a time three hours in the
+        # past. It will return a count of 3 -- the sum of the return
+        # values from our mocked _update_circulation().
+        now = datetime.datetime.utcnow()
+        one_hour_ago = now - datetime.timedelta(hours=1)
+        three_hours_ago = now - datetime.timedelta(hours=3)
+        monitor = Mock(self._db, self.collection, api_class=MockEnkiAPI)
+        eq_(3, monitor.update_circulation(three_hours_ago))
+
+        # slice_timespan() sliced up the timeline into two-hour
+        # chunks. It yielded up two chunks: "three hours ago" to "one
+        # hour ago" and "one hour ago" to "now".
+        #
+        # _update_circulation() was called on each chunk, and in each
+        # case the return value was an item popped from monitor.sizes.
+        chunk1, chunk2 = monitor._update_circulation_called_with
+        eq_((three_hours_ago, one_hour_ago), chunk1)
+        eq_(one_hour_ago, chunk2[0])
+        assert (chunk2[1] - now).total_seconds() < 2
```

Comment:
Is this because it's hard to test that the "now" from the test is the current "now?

Reply:
It's impossible -- the two times will always be slightly off. So yeah, I just check that they're basically the same time.